lucene/analysis/common/src/test/org/apache/lucene/analysis/hunspell/TestsFromOriginalHunspellRepository.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.analysis.hunspell;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.ParseException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * Same as {@link SpellCheckerTest}, but checks all Hunspell's test data. The path to the checked
+ * out Hunspell repository should be in {@code -Dhunspell.repo.path=...} system property.
+ */
+@RunWith(Parameterized.class)
```

Comment:
Please run with the default runner (extend RandomizedTest or, better, LuceneTestCase) - it adds a number of extra checks to make sure no interference between tests, proper reporting, etc. If you wish to parameterize via constructor, you still can - an example is here:

https://github.com/randomizedtesting/randomizedtesting/blob/master/examples/maven/src/main/java/com/carrotsearch/examples/randomizedrunner/Test007ParameterizedTests.java

Reply:
Since this test isn't intended to be run on CI (at least for now), I thought it's safe to skip those. Apparently it's not? :)

Reply:
If you leave this behind, it'll just be there forever... I can convert it if you wish. And I think it does make sense to maybe run these tests on a CI at some point. Just to make sure no regressions occur.

Reply:
I'll deal with it. Just asking :)

Reply:
You can also assumeTrue(false) (throw an assumption exception) if the property is not defined. This will mark the test as ignored rather than empty?

Reply:
I'll check if assume works in parameters method, thanks!

Unfortunately randomized+parameterized doesn't play nicely with IntelliJ. First, the presentation is pretty verbose:

![image](https://user-images.githubusercontent.com/122009/106443383-b4625e80-647c-11eb-8fa4-973ddd5f99cb.png)

vs

![image](https://user-images.githubusercontent.com/122009/106443464-ccd27900-647c-11eb-8029-13556518dd6b.png)

Second, more importantly, it's impossible to right-click a single test and run it separately, while that works with the standard parameterized runner. Are you aware of ways to improve that?

Reply:
Test names are essentially Descriptions in JUnit. Filtering by the same test description should re-run the same test. IntelliJ's behavior has been changing over time here and I really don't know why it's not working now... Do you have it on that branch? I'll take a look (later).

Reply:
Yes, I've merged it with the master and renamed tests. The last commit changes the parameterization to the recommended, and that's where I get that behavior :(

Reply:
Thanks. I'll take a look later, Peter. Will get back to you.

Reply:
I checked intellij and parameterized tests tonight. It's what I was afraid of - test descriptions are emitted correctly (in my opinion) but they're *interepreted* differently depending on the tool (and time when you check...). 

The reason why you see the class name and test method before each actual test is because these supposedly "hidden" elements allowed tools to go back to the source code of a test with an arbitrary name (if you double-click on a test in IntelliJ it will take you back to the test method). Relaunching of a single test must have changed at some point because it used to be an exact name filter... but now I it just reruns all tests under a test method (all parameter variations).

It's worth mentioning that this isn't consistent even in IntelliJ itself - if I run a simple(r) parameterized test via IntelliJ launcher, I get this test suite tree:

![image](https://user-images.githubusercontent.com/199470/106523393-561b9700-64e1-11eb-9000-4c4a66117331.png)

But when I run the same test via gradle launcher (from within the IDE), I get this tree:

![image](https://user-images.githubusercontent.com/199470/106523340-3e441300-64e1-11eb-958f-dc36ece73d66.png)

I don't know if there is a way to make all the tools happy; test descriptions and nesting is broken in JUnit 4.x itself.

Given the above, please feel free to revert back to what works for you. I'd name the test class TestHunspellRepositoryTestCases for clarity. Also, this test will not run under Lucene test framework because the security manager won't let you access arbitrary paths outside the build location. You'd need to add this to tests.policy:
```
permission java.io.FilePermission "${hunspell.repo.path}${/}-", "read";
```
Don't know whether it's worth it at the moment though.

Reply:
Filed an issue for myself here: https://github.com/randomizedtesting/randomizedtesting/issues/295.

Reply:
Thanks! Now I'm starting to doubt whether this approach makes sense at all. I could avoid parameterization by generating test methods explicitly by files, with some risk that new files appear (which could be checked by additional code).

And is it OK to modify the test policy for such local tests? I planned to add more not-easy-to-have-in-CI tests, which would measure performance and check correctness. They'd need external files with dictionaries, corpora for various languages (external or is there anything internal already?), and a test-only Hunspell JNI library for comparison (which needs a native binary and a couple of other jars, all of them need sha and license files, and it all gets quite verbose). Do you think the benefits of having this in the repo outweigh the costs? I could also leave this all locally, since I seem to be the only one needing these tests in the near future.

Reply:
I think they should reside in the repo if they are useful (even for local launches). What I'm afraid of is that if these tools are not in use, they'll eventually degrade and stop working without anyone noticing. 

I think the way to integrate such tests properly would be to add a specific gradle test task which would configure an appropriate policy, require pointers to the required resources, etc. This way these tests can be run as a CI run (somewhere... maybe a github action, even?).

I think this can be ironed out later on, once you've written (notice the 'you' here... ;) more of such tests - the patterns of making them work with the CI will naturally emerge from that.

For now, feel free to use that original parameterized test runner - I'll look into making IntelliJ work with randomizedtesting again (because I use it here and in other projects). It's a moving target and thus a bit discouraging (I did the same thing a few times in the past already for various IDEs that interpreted test descriptions differently).